### PR TITLE
Pom dependency/version updates for final release

### PR DESCRIPTION
**Pull request by svanoort**
Bump the versions, pull in the latest releases and generally clean up All The Things some.

### CODE REVIEW

**Comment by svanoort** [comment]
@reviewbybees

File: pom.xml [hunk 1]
```diff
@@ -52,13 +52,12 @@
...
             <groupId>org.jenkins-ci.plugins.workflow</groupId>
             <artifactId>workflow-api</artifactId>
-            <version>2.2-beta-2</version> <!-- allows consuming the new APIs -->
+            <version>2.2</version> <!-- allows consuming the new APIs -->
         </dependency>
         <dependency>
             <groupId>org.jenkins-ci.plugins</groupId>
             <artifactId>pipeline-input-step</artifactId>
             <version>2.0</version>
-            <optional>true</optional>
```
Comment (svanoort) [inline pom.xml]: We're tied to the 2.0 APIs now, this is to ensure they're pulled in (potential fix for some dependency issues too).

**Comment by svanoort** [comment]
Note:  still keeping beta-version dependency on the block-scoped stage step until it sees full release.

**Comment by abayer** [comment]
:bee:

**Comment by jglick** [comment]
:bee:

**Comment by svanoort** [comment]
Thanks guys!

**Pull request merged**